### ISSUE TITLE
Is there an option to use Kaniko debug image without login?

### ISSUE THREAD

**Issue by YevheniiPokhvalii** [GoogleContainerTools/kaniko#2622]
When I use Kaniko on OpenShift, I don't need to log in to the internal OpenShift registry. Authentication is only based on IRSA.

The slim executor Kaniko image works fine since it does not contain any authentication binaries and just pushes an image to the internal OpenShift registry.

It seems that it is not possible to use the Kaniko debug executor image with the internal OpenShift registry since it always tries to log in to the internal OpenShift registry and throws an error.

Are there any workarounds with the debug image? 
If not, please consider adding the `--no-login` option to the Kaniko executor.

**Comment by aaron-prindle** [comment]
Thanks for flagging this issue @YevheniiPokhvalii.  Can you clarify what the `--no-login` flag should do?  Is the feature request here to allow the ability to have Kaniko not-auth with the registry that is added at `--destination=...` (currently part of it's validation)?

**Comment by YevheniiPokhvalii** [comment]
> Thanks for flagging this issue @YevheniiPokhvalii. Can you clarify what the `--no-login` flag should do? Is the feature request here to allow the ability to have Kaniko not-auth with the registry that is added at `--destination=...` (currently part of it's validation)?

Yes, I think you got me right. Here is the script we use in a Tekton pipeline:

```
      script: |
        /kaniko/executor \
          --dockerfile=$(workspaces.source.path)/$(params.DOCKERFILE) \
          --context=$(workspaces.source.path)/$(params.CONTEXT) \
          --destination=$(params.IMAGE) \
          --digest-file=$(results.IMAGE_DIGEST.path) \
          --cache=true \
          --cache-repo=$(params.KANIKO_CACHE_REPO) \
          --tar-path=$(params.IMAGE_TAR).tar
```

This script works with the remote registry and does not work with the internal OpenShift registry because it does not require login.
It might help to avoid the described issue if there is an option to use the `--no-login` flag.

Probably, if I unset this variable `DOCKER_CONFIG`, the issue will also be fixed. But it needs testing and I currently don't the opportunity to check it: https://github.com/GoogleContainerTools/kaniko/blob/main/deploy/Dockerfile#L84

To sum up, the `--no-login` feature may be useful with the internal registries.

**Comment by aaron-prindle** [comment]
Recently a flag was added which skips the initial authentication step that kaniko does:
--skip-push-permission
https://github.com/GoogleContainerTools/kaniko/pull/2657

Additionally there are 3 related `--insecure-*` flags for kaniko that allow it to interact with an insecure registry w/o auth:
https://github.com/GoogleContainerTools/kaniko#flag---insecure-registry

**Comment by YevheniiPokhvalii** [comment]
Thanks! Yep, push was needed in my case. Unfortunately, I can't test it now since I no longer work on this project with OpenShift.
